src/main.ts

```diff
@@ -0,0 +1,146 @@
+const githubEventPayload = require(githubEventPayloadFile);
+const triggerEventName = process.env.GITHUB_EVENT_NAME;
+
+export const parsedNotificationMessage = function parseEventToMessage(eventPayloadText: string): NotificationMessage | void {
+    let account;
+    let parsedSchema;
+    try {
+        let eventName: string = triggerEventName || 'event';
+        if (core.getInput('event_id')) {
+            eventName = core.getInput('event_id');
+        }
+
+        switch (eventName) {
+            case EventType.PUSH:
+                parsedSchema = schemaOnpush.validateSync(eventPayloadText);
+                account = parsedSchema.pusher.name;
+                return {
+                    message: `**Commit to GitHub** by ${account}`,
+                    url: parsedSchema.compare,
+                    details: `Comment: ${parsedSchema.head_commit.message}`
+                };
+            case EventType.PULL_REQUEST:
+                parsedSchema = schemaOnPullRequest.validateSync(eventPayloadText);
+                account = parsedSchema.pull_request.user.login;
+                return {
+                    message: `**PR submitted to Github** by ${account} with title: ${parsedSchema.pull_request.title}`,
+                    url: parsedSchema.pull_request.html_url,
+                    details: `Pr for merging ref ${parsedSchema.pull_request.head.ref} to base branch ${parsedSchema.pull_request.base.ref}`
+                };
+            case EventType.ISSUE:
+                parsedSchema = schemaOnIssue.validateSync(eventPayloadText);
+                account = parsedSchema.sender.login;
+                return {
+                    message: `**New/updated GitHub issue**: ${parsedSchema.issue.title}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            case EventType.ISSUE_COMMENT:
+                parsedSchema = schemaOnIssueComment.validateSync(eventPayloadText);
+                return {
+                    message: `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`,
+                    url: parsedSchema.issue.html_url,
+                    details: `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`
+                };
+            default:
+                return {
+                    message: eventName ? eventName.replace(/_/g, '.') : 'A GitHub Actions event has occurred',
+                    url: `https://github.com/${repository}/actions`,
+                    details: core.getInput('details')
+                };
+        }
+    } catch (error) {
+        let errorDetails;
+        if (error instanceof ValidationError) {
+            errorDetails = error.errors;
+        } else {
+            errorDetails = error.message;
+        }
+        return core.setFailed(`ERROR : ${errorDetails}`);
```

Comment:
looks like on line 138 you rely on this to return a falsy value (undefined probably) - but are you sure `core.setFailed` returns undefined?

You can make it more explicit either by throwing an error, or something like that:

```suggestion
        return void core.setFailed(`ERROR : ${errorDetails}`);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void